Framework/DataObjects/src/Workspace2D.cpp

```diff
@@ -68,19 +68,19 @@ void Workspace2D::init(const std::size_t &NVectors, const std::size_t &XLength,
   m_noVectors = NVectors;
   data.resize(m_noVectors);
 
-  MantidVecPtr t1, t2;
-  t1.access().resize(XLength); // this call initializes array to zero
-  t2.access().resize(YLength);
+  auto x = Kernel::make_cow<HistogramData::HistogramX>(XLength);
+  HistogramData::Counts y(YLength);
+  HistogramData::CountStandardDeviations e(YLength);
   for (size_t i = 0; i < m_noVectors; i++) {
     // Create the spectrum upon init
-    auto spec = new Histogram1D();
+    auto spec =
+        new Histogram1D(HistogramData::getHistogramXMode(XLength, YLength));
     data[i] = spec;
     // Set the data and X
-    spec->setX(t1);
-    spec->setDx(t1);
-    spec->resetHasDx();
+    spec->setX(x);
     // Y,E arrays populated
-    spec->setData(t2, t2);
+    spec->setCounts(y);
```

Comment:
Counts and errors now have distinct underlying types (`HistogramY` and `HistogramE`), so they cannot be shared any more.

On the plus side, we now have getters and setter for the shared pointers (or rather cow pointers) for `Y` and `E`. Those did not exist in the old interface. We should thus be able to improve the overall amount of sharing.